stellargraph/data/unsupervised_sampler.py

```diff
@@ -153,9 +153,14 @@ def run(self, batch_size):
             ]
         )
 
+        positive_pairs = self.graph.node_ids_to_ilocs(positive_pairs.flatten()).reshape(
```

Comment:
My thinking was: 

- 'RandomWalks` have to return node ids to be compatible with our gensim work flows
- 'GraphWalks` are contructed by us in `BatchedNodeGenerator` and  `BatchedLinkGenerator`, so we can keep the API and still use `node_ilocs` internally
- `UnsupervisedSampler` is created by the user and feed `node_ilocs` to a tensorflow model, so there's no way for us to set `node_ilocs=True` on `UnsupervisedSampler`

It would be upto the user to set `use_ilocs=True` on `UnsupervisedSampler`, which yields a 2x speedup over `use_ilocs=False` and a 1.5x speedup over develop. And I think most users wouldn't specify this option! What do you think, should we break the API in this case/do you have any ideas on how to do this in a way that doesn't break the API?

I'll do more timing and post the results by this change but for iterating the citeseer `Attri2Vec` demo `OnDemandLinkSequence` takes:

- 640ms on this PR
-940ms on develop
- 1.2s on #1220

Reply:
That all makes sense 👍 

> RandomWalks have to return node ids to be compatible with our gensim work flows

FWIW (as you might realise), this isn't even quite enough, given our graphs can have non-string IDs, but gensim requires strings. One option (for future work) would be to return `str`'d ilocs in the random walkers... another would be to do the embeddings with ilocs directly (maybe on top of #536).